### PR TITLE
blog post: Deploy on Kubernetes with Helm

### PR DESCRIPTION
**Pull request by marcosgopen**
This blog should aid ease of use of the Helm Wildfly chart (https://github.com/wildfly/wildfly-charts/[https://github.com/wildfly/wildfly-charts/) to deploy a simple Wildfly application.

### CODE REVIEW

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+$ helm search repo wildfly
+NAME                  	CHART VERSION	APP VERSION	DESCRIPTION                                       
+wildfly/wildfly       	2.3.1        	           	Build and Deploy WildFly applications on OpenShift
+wildfly/wildfly-common	2.1.0        	           	A library chart for WildFly-based applications  
+----
+
+## Make sure your application pom contains the wildfly-maven-plugin and maven-war-plugin plugins 
+
+In our helloworld quickstart the maven-war-plugin is already present (recommanded version is 3.2.3).
+Please be sure to have an updated version of it under the plugins section of the pom:
```
[the remark this answers is not in the document]
Comment (marcosgopen) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc]: Thanks @yersan for your review! You are right, I deleted the part related to maven-war-plugin and I added a note to clarify that point. The blog is now simpler, thanks.

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+55ea6d5a354e: Pushed 
+latest: digest: sha256:1d72e7b4f8dd75414a2b1dc8da3505906b0211ac6962e871b84f2c8ead43cb3a size: 1377
+----
+Now your image is available in your container registry (in my case it is available at https://quay.io/repository/rh_ee_msappegr/helloworld)
+
+Note: make sure your registry is public in order to pull it from the cluster. Otherwise you will need to create a https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/[docker-registry secret] .
+
+## Create the YAML value file under the charts folder
+
+In our example the charts folder already exists under quickstart/helloworld/charts, so move in that directory and create a file named values.yaml
```
Comment (bstansberry) [inline _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:272]: s/in that/to

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+        </profile>
+...
+    </profiles>    
+----
+
+Note: In our helloworld quickstart the maven-war-plugin is used to create the war. If you want to use a different plugin to create the war or jar you need to set the 'filename' configuration of the wildfly-maven-plugin plugin accordingly.
+
+## Build the docker image with the wildfly-maven-plugin plugin 
+
+Note: In this example we use 'docker', but 'podman' works as well.
```
Comment (bstansberry) [inline _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:107]: How does this note apply to the subsequent block? I see in the log output that the wildfly-maven-plugin is using docker, but how does the user use podman?

I'm pretty sure the answer is podman aliases the docker commands so it just works, in which case perhaps say:

Note: In this example the wildfly-maven-plugin calls 'docker', but if you have 'podman' installed that will work as well.

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+
+[source]
+----
+image:
+  name: quay.io/rh_ee_msappegr/helloworld # change it with the image you have just pushed
+build:
+  enabled: false # The build part is not needed since we have already built our application with the wildfly-maven-plugin plugin
+deploy:
+  route:
+    enabled: false # the route could be enabled only for Openshift  clusters
```
Comment (bstansberry) [inline _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:290]: Maybe

'the route can be enabled, but only for OpenShift clusters'

For sure 

s/Openshift/OpenShift

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+    <head>
+        <meta http-equiv="Refresh" content="0; URL=HelloWorld">
+    </head>
+</html>
+----
+
+The application is up and running.
+
+## Conclusion
+It is possible to run you application with very simple steps.
```
Comment (bstansberry) [inline _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:381]: s/you/your

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+
+All is necessary is:
```
Comment (bstansberry) [inline _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:383]: s/All is/All that is/g

Comment (bstansberry) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:383]: Also, I suggest

s/is:/is to:/g

and then drop the 'to' at the beginning of the 3 bullets.

That reads more naturally to me, but YMMV.

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+synopsis: How to deploy WildFly applications on Kubernetes with Helm.
+author: marcosgopen
+---
+
+https://helm.sh[Helm] is a tool that helps define, install and upgrade complex applications on Kubernetes clusters.
+
+In this post, we will use the Wildfly chart from https://github.com/wildfly/wildfly-charts/[https://github.com/wildfly/wildfly-charts/] to deploy a simple Wildfly application:
+
+* Create a simple Java application
+* use the wildfly-maven-plugin's image goal to create a Docker image straight from Maven
```
[the remark this answers is not in the document]
Comment (yersan) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:15]: I also suggest starting all bullet phrases with the first letter as an upper case, not only the first one (Create a simple ....)

* Create a simple ...
* Use the ...
* Push that image ...
* Use Helm to ...

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+...
+    </profiles>    
+----
+
+Note: In our helloworld quickstart the maven-war-plugin is used to create the war. If you want to use a different plugin to create the war or jar you need to set the 'filename' configuration of the wildfly-maven-plugin plugin accordingly.
```
Comment (yersan) [inline _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:103]: I think you can remove this sentence completely, indeed I've realized there is no explicit `maven-war-plugin` declaration in the `helloworld/pom.xml`. In my opinion, this note here doesn't add value the the blog post goal

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+
+## Build the docker image with the wildfly-maven-plugin plugin 
+
+Note: In this example we use 'docker', but 'podman' works as well.
+
+[source,nowrap]
+----
+ $ mvn package
+ $ mvn wildfly:image -Popenshift
```
Comment (yersan) [inline _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:112]: The `package` goal must also be executed under the -Popenshift profile

I would suggest using `$ mvn package wildfly:image -Popenshift`

Comment (marcosgopen) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:112]: Good catch! Thanks. I will use `$ mvn package wildfly:image -Popenshift` so I am going to update also the output.

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc (hunk 8 above)
Comment (marcosgopen) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc:103]: I agree! Thanks.

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (marcosgopen) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc]: Good suggestions, thanks!

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc (hunk 2 above)
Comment (marcosgopen) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc]: Thank you very much for your review, I substituted `in that` to `to that`. I hope it is what you meant.

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc (hunk 4 above)
Comment (marcosgopen) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc]: Thank you, I addressed your suggestion.

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc (hunk 5 above)
Comment (marcosgopen) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc]: Thanks

File: _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc (hunk 6 above)
Comment (marcosgopen) [reply _posts/2023-06-16-deploy-on-kubernetes-with-helm.adoc]: Thank you, I addressed your suggestions.

**Review by yersan — APPROVED** [review]
LGTM

**Comment by marcosgopen** [comment]
@bstansberry Could you please re-review this PR and merge it if it's ready? Thanks

**Review by bstansberry — APPROVED** [review]
Thanks @marcosgopen. My local ruby env was broken so I couldn't validate this rendered ok, but I got it sorted today and it renders fine.

**Pull request merged**